src/v/cloud_storage/tests/cache_test.cc

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "bytes/iobuf.h"
+#include "cache_test_fixture.h"
+#include "cloud_storage/cache_service.h"
+#include "test_utils/fixture.h"
+#include "units.h"
+
+#include <seastar/core/seastar.hh>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace cloud_storage;
+
+FIXTURE_TEST(put_creates_file, cache_test_fixture) {
+    ss::sstring data_string;
+    while (data_string.length() < 4096) {
+        data_string += SAMPLE_STRING1;
+    }
+    iobuf buf;
+    buf.append(data_string.data(), data_string.length());
+
+    cache_service.put(KEY, make_iobuf_input_stream(std::move(buf))).get();
+
+    bool file_exists = ss::file_exists((CACHE_DIR / KEY).native()).get();
+    BOOST_CHECK(file_exists);
+}
+
+FIXTURE_TEST(get_after_put, cache_test_fixture) {
+    ss::sstring data_string;
+    while (data_string.length() < 4096) {
+        data_string += SAMPLE_STRING1;
+    }
+    iobuf buf;
+    buf.append(data_string.data(), data_string.length());
+
+    cache_service.put(KEY, make_iobuf_input_stream(std::move(buf))).get();
+
+    std::optional<cloud_storage::cache_item> returned_item
+      = cache_service.get(KEY).get();
+    BOOST_REQUIRE(returned_item);
+    BOOST_CHECK_EQUAL(returned_item->size, data_string.length());
+    BOOST_CHECK_EQUAL(
+      returned_item->body.read_exactly(data_string.length()).get().get(),
+      data_string.data());
+    BOOST_CHECK(!returned_item->body.read().get().get());
```

Comment:
A `const char*` comparison still kind of bothers me - `temporary_buffer`s returned from `read_exactly` are not guaranteed to be valid zero-terminated strings. I think we can use a string_view for comparison here and in the next test.

Reply:
fixed to use `string_view`